stress: failed test in cockroach/sql/sql.test: TestRaceWithBackfill

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/2cc5a467f6c280e97e49a22b884b649fd7790273

Stress build found a failed test:

```
=== RUN   TestRaceWithBackfill
W160517 07:06:54.874360 gossip/gossip.go:893  not connected to cluster; use --join to specify a connected node
I160517 07:06:54.874990 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
W160517 07:06:54.876153 gossip/gossip.go:893  not connected to cluster; use --join to specify a connected node
I160517 07:06:54.876321 server/node.go:401  store store=0:0 ([]=) not bootstrapped
I160517 07:06:54.877515 storage/replica_command.go:1470  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m55.877308039s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:54.877456484 +0000 UTC]
I160517 07:06:54.878010 server/node.go:327  **** cluster {4780cd5b-ed50-487f-947c-491a59718743} has been created
I160517 07:06:54.878034 server/node.go:328  **** add additional nodes by specifying --join=127.0.0.1:44223
W160517 07:06:54.878050 gossip/gossip.go:893  not connected to cluster; use --join to specify a connected node
I160517 07:06:54.878672 server/node.go:414  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:8209195008 RangeCount:0}
I160517 07:06:54.878698 server/node.go:302  node ID 1 initialized
I160517 07:06:54.878750 storage/stores.go:287  read 0 node addresses from persistent storage
I160517 07:06:54.878815 server/node.go:535  connecting to gossip network to verify cluster ID...
I160517 07:06:54.878938 server/node.go:556  node connected via gossip and verified as part of cluster {"4780cd5b-ed50-487f-947c-491a59718743"}
I160517 07:06:54.878973 server/node.go:355  [node=1] Started node with [[]=] engine(s) and attributes []
I160517 07:06:54.879026 server/server.go:399  starting https server at 127.0.0.1:43726
I160517 07:06:54.879046 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160517 07:06:54.879201 server/server.go:400  starting grpc/postgres server at 127.0.0.1:44223
I160517 07:06:54.879541 server/updates.go:155  No previous updates check time.
I160517 07:06:54.881589 storage/replica_command.go:2011  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160517 07:06:54.885512 storage/replica_command.go:1470  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m55.978360814s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:54.885469278 +0000 UTC]
I160517 07:06:54.885606 storage/replica_command.go:2011  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160517 07:06:54.922589 storage/replica_command.go:1470  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m55.978360814s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:54.922549583 +0000 UTC]
I160517 07:06:54.922672 storage/replica_command.go:2011  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
I160517 07:06:54.935456 storage/replica_command.go:1470  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m55.985296972s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:54.935424247 +0000 UTC]
I160517 07:06:54.935532 storage/replica_command.go:2011  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160517 07:06:54.984737 storage/replica_command.go:1470  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m56.034537705s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:54.984694316 +0000 UTC]
I160517 07:06:54.984851 storage/split_queue.go:100  splitting range=5 [/Table/14-/Max) at keys [/Table/50/0]
I160517 07:06:54.985165 storage/replica_command.go:2011  initiating a split of range=5 [/Table/14-/Max) at key /Table/50
I160517 07:06:55.004929 storage/replica_command.go:1470  range 6: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m56.054677348s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:55.004880216 +0000 UTC]
I160517 07:06:55.005105 storage/split_queue.go:100  splitting range=6 [/Table/50-/Max) at keys [/Table/51/0]
I160517 07:06:55.005496 storage/replica_command.go:2011  initiating a split of range=6 [/Table/50-/Max) at key /Table/51
I160517 07:06:55.877212 gossip/gossip.go:919  starting client to 127.0.0.1:44223
I160517 07:06:55.877761 gossip/client.go:86  closing client to node 1 (127.0.0.1:44223): gossip/client.go:180: stopping outgoing client to node 1 (127.0.0.1:44223); loopback connection
I160517 07:06:56.040496 storage/replica_command.go:1470  range 7: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406519h6m57.090200331s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-17 07:06:56.040447247 +0000 UTC]
I160517 07:06:56.416756 sql/lease.go:224  publish (count leases): descID=51 name=test version=4 count=1
I160517 07:06:56.464742 sql/lease.go:224  publish (count leases): descID=51 name=test version=5 count=1
I160517 07:06:56.900174 sql/lease.go:224  publish (count leases): descID=51 name=test version=6 count=1
I160517 07:06:57.099131 sql/lease.go:224  publish (count leases): descID=51 name=test version=7 count=1
I160517 07:06:57.158610 sql/lease.go:224  publish (count leases): descID=51 name=test version=8 count=1
I160517 07:06:57.554581 sql/lease.go:224  publish (count leases): descID=51 name=test version=9 count=1
I160517 07:06:57.706500 sql/lease.go:224  publish (count leases): descID=51 name=test version=10 count=1
I160517 07:06:57.766619 sql/lease.go:224  publish (count leases): descID=51 name=test version=11 count=1
W160517 07:06:58.756545 sql/schema_changer.go:229  node unavailable; try another peer
W160517 07:06:58.756643 sql/schema_changer.go:193  node unavailable; try another peer
--- FAIL: TestRaceWithBackfill (3.89s)
	schema_changer_test.go:469: schema change ALTER TABLE t.test ADD COLUMN x DECIMAL DEFAULT (DECIMAL '1.4') took 525.089925ms
	schema_changer_test.go:469: schema change ALTER TABLE t.test DROP pi took 490.363878ms
	schema_changer_test.go:522: pq: duplicate key value (v)=(4009) violates unique constraint "vidx\n```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
0 runs so far, 0 failures, over 35s
0 runs so far, 0 failures, over 40s
14 runs so far, 0 failures, over 45s
15 runs so far, 0 failures, over 50s
15 runs so far, 0 failures, over 55s
15 runs so far, 0 failures, over 1m0s
15 runs so far, 0 failures, over 1m5s
15 runs so far, 0 failures, over 1m10s
15 runs so far, 0 failures, over 1m15s
15 runs so far, 0 failures, over 1m20s
22 runs so far, 0 failures, over 1m25s
28 runs so far, 0 failures, over 1m30s
30 runs so far, 0 failures, over 1m35s
30 runs so far, 0 failures, over 1m40s
30 runs so far, 0 failures, over 1m45s
30 runs so far, 0 failures, over 1m50s
30 runs so far, 0 failures, over 1m55s
30 runs so far, 0 failures, over 2m0s
33 runs so far, 0 failures, over 2m5s
40 runs so far, 0 failures, over 2m10s
45 runs so far, 0 failures, over 2m15s
45 runs so far, 0 failures, over 2m20s
45 runs so far, 0 failures, over 2m25s
45 runs so far, 0 failures, over 2m30s
45 runs so far, 0 failures, over 2m35s
45 runs so far, 0 failures, over 2m40s
47 runs so far, 0 failures, over 2m45s
53 runs so far, 0 failures, over 2m50s
58 runs so far, 0 failures, over 2m55s
60 runs so far, 0 failures, over 3m0s
60 runs so far, 0 failures, over 3m5s
60 runs so far, 0 failures, over 3m10s
60 runs so far, 0 failures, over 3m15s
60 runs so far, 0 failures, over 3m20s
60 runs so far, 0 failures, over 3m25s
67 runs so far, 0 failures, over 3m30s
72 runs so far, 0 failures, over 3m35s
74 runs so far, 0 failures, over 3m40s
75 runs so far, 0 failures, over 3m45s
75 runs so far, 0 failures, over 3m50s
75 runs so far, 0 failures, over 3m55s
75 runs so far, 0 failures, over 4m0s
75 runs so far, 0 failures, over 4m5s
77 runs so far, 0 failures, over 4m10s
82 runs so far, 0 failures, over 4m15s
85 runs so far, 0 failures, over 4m20s
89 runs so far, 0 failures, over 4m25s
89 runs so far, 0 failures, over 4m30s
89 runs so far, 0 failures, over 4m35s
89 runs so far, 0 failures, over 4m40s
89 runs so far, 0 failures, over 4m45s
92 runs so far, 0 failures, over 4m50s
94 runs so far, 0 failures, over 4m55s
99 runs so far, 0 failures, over 5m0s
103 runs so far, 0 failures, over 5m5s
103 runs so far, 0 failures, over 5m10s
103 runs so far, 0 failures, over 5m15s
103 runs so far, 0 failures, over 5m20s
103 runs so far, 0 failures, over 5m25s
104 runs completed, 1 failures, over 5m27s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/sql/delete.go
+++ b/sql/delete.go
@@ -141,6 +141,10 @@ func (d *deleteNode) Next() bool {
 	}
 
 	if !d.run.rows.Next() {
+		if d.run.err = d.run.rows.Err(); d.run.err != nil {
+			d.run.done = true
+			return false
+		}
 		// We're done. Finish the batch.
 		d.run.err = d.tw.finalize()
 		d.run.done = true

--- a/sql/insert.go
+++ b/sql/insert.go
@@ -276,6 +276,10 @@ func (n *insertNode) Next() bool {
 	}
 
 	if !n.run.rows.Next() {
+		if n.run.err = n.run.rows.Err(); n.run.err != nil {
+			n.run.done = true
+			return false
+		}
 		// We're done. Finish the batch.
 		n.run.err = n.tw.finalize()
 		n.run.done = true

--- a/sql/update.go
+++ b/sql/update.go
@@ -310,9 +310,12 @@ func (u *updateNode) Next() bool {
 	}
 
 	if !u.run.rows.Next() {
+		if u.run.err = u.run.rows.Err(); u.run.err != nil {
+			u.run.done = true
+			return false
+		}
 		// We're done. Finish the batch.
-		err := u.tw.finalize()
-		u.run.err = err
+		u.run.err = u.tw.finalize()
 		u.run.done = true
 		return false
 	}

